app/src/main/java/org/twinone/irremote/providers/common/CommonProviderFragment.java

```diff
@@ -173,21 +175,35 @@ public void onItemClick(AdapterView<?> parent, View view, int position,
         }
     }
 
+    /**
+     * Build a remote from our built-in database.
+     * Either from JSON definition or from button files.
+     */
     private Remote buildRemote() {
-        Remote r = new Remote();
-        r.name = mTarget.deviceName + " " + mTarget.deviceType;
         final String remotedir = getDBPath() + File.separator + mTarget.deviceName;
-        for (String name : listAssets(remotedir)) {
-            int id = Integer.parseInt(name.substring(2).split("\\.")[0]);
-            Button b = new Button(id);
-            b.code = FileUtils.read(getActivity().getAssets(), remotedir
-                    + File.separator + name);
-            b.text = ComponentUtils.getCommonButtonDisplayName(b.id,
-                    getActivity());
-            r.addButton(b);
+
+
+        String[] assets = listAssets(remotedir);
+        if (assets.length == 1 && Objects.equals(assets[0], "remote.json")) {
+            // Create a remote from JSON
+            String json = FileUtils.read(getActivity().getAssets(), remotedir + File.separator + assets[0]);
+            return Remote.deserialize(json);
+        } else {
```

Comment:
This else statement may not be required. The 'if' section already returns when json exits.